### PR TITLE
Meta: config and build cleanup

### PR DESCRIPTION
**Pull request by fregante**
- XO’s configuration grew a lot, so I extracted it from package.json
- Instead of using a webpack plugin for asynchronous type checking, I just run `tsc` in parallel and get native error messages
- Webpack’s config file could be improved slightly
- I had also tried avoiding duplicated code with [SplitChunks](https://webpack.js.org/plugins/split-chunks-plugin/) but for some reason the content scripts weren't executing (maybe due to https://github.com/webpack/webpack/issues/7230). Also Chrome doesn't like `~` in filenames, but that was an [easy fix](https://webpack.js.org/plugins/split-chunks-plugin/#splitchunksautomaticnamedelimiter)

### CODE REVIEW

File: .xo-config.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+				svg: 'always'
+			}
+		],
+		'unicorn/no-fn-reference-in-iterator': 'off'
+	},
+	settings: {
+		react: {
+			version: '16.3'
+		}
+	}
```
Comment (sindresorhus) [inline .xo-config.js:59]: https://github.com/xojs/eslint-config-xo-react/blob/ab2d7bb616c59b48d8b2f9aa1a89723deceaccaf/index.js#L12-L16

File: .xo-config.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		'import/no-anonymous-default-export': 'error',
+		'import/no-commonjs': 'error',
+		'import/no-named-default': 'error',
+		'import/no-unassigned-import': 'off',
+		'import/prefer-default-export': 'error',
+		'import/extensions': [
+			'error',
+			'never',
+			{
+				svg: 'always'
```
Comment (sindresorhus) [inline .xo-config.js:50]: Happy to have this one added to XO.

File: .xo-config.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		'@typescript-eslint/prefer-readonly-parameter-types': 'off',
+		'no-alert': 'off',
+		'no-void': 'off',
+		'react/function-component-definition': [
+			'error',
+			{
+				namedComponents: 'function-declaration'
+			}
+		],
+		'react/jsx-first-prop-new-line': 'error',
```
Comment (sindresorhus) [inline .xo-config.js:25]: https://github.com/xojs/eslint-config-xo-react/blob/ab2d7bb616c59b48d8b2f9aa1a89723deceaccaf/index.js#L101

File: .xo-config.js (hunk 1 above)
Comment (fregante) [reply .xo-config.js:59]: We don't have react so the detection fails

**Pull request merged**